frontend/src/screens/wallet/Swaps.tsx

```diff
@@ -0,0 +1,499 @@
+          <RadioGroupItem value="internal" id="internal" className="shrink-0" />
+          <Label
+            htmlFor="internal"
+            className="text-primary font-medium cursor-pointer"
+          >
+            Spending balance
+          </Label>
+        </div>
+        <div className="flex items-start space-x-2">
+          <RadioGroupItem value="external" id="external" className="shrink-0" />
+          <Label
+            htmlFor="external"
+            className="text-primary font-medium cursor-pointer"
+          >
+            External lightning wallet
+          </Label>
+        </div>
+      </RadioGroup>
+      {!isInternalSwap && (
+        <div className="grid gap-1.5">
+          <Label>Receiving lightning address</Label>
+          <div className="flex gap-2 mb-4">
+            <Input
+              placeholder="satoshi@getalby.com"
+              value={destination}
+              onChange={(e) => setDestination(e.target.value)}
+              required
+            />
+            <Button
+              type="button"
+              variant="outline"
+              className="px-2"
+              onClick={paste}
+            >
+              <ClipboardPasteIcon className="w-4 h-4" />
+            </Button>
+          </div>
+        </div>
+      )}
+      {/* TODO: Review fee for swap ins */}
+      <div className="flex items-center justify-between border-t py-4">
+        <Label>Fee</Label>
+        <p className="text-muted-foreground text-sm">
+          {swapsSettings.albyServiceFee + swapsSettings.boltzServiceFee}% +
+          on-chain fees
+        </p>
+      </div>
+      <LoadingButton loading={loading}>Swap In</LoadingButton>
+    </form>
+  );
+}
+
+function SwapOutForm() {
+  const { toast } = useToast();
+  // TODO: Optimize by setting this from the backend
+  const { data: onchainAddress } = useOnchainAddress();
+  const { data: swapsSettings, mutate } = useSwaps();
+
+  const [swapTo, setSwapTo] = useState("internal");
+  const [balanceThreshold, setBalanceThreshold] = useState("");
+  const [swapAmount, setSwapAmount] = useState("");
+  const [destination, setDestination] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  const [recurringSwaps, setRecurringSwaps] = useState(false);
+
+  const onSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    try {
+      setLoading(true);
+      await request(`/api/wallet/${recurringSwaps ? "swaps" : "swap"}`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          swapType: "out",
+          swapAmount: parseInt(swapAmount),
+          destination: swapTo === "internal" ? onchainAddress : destination,
+          balanceThreshold: parseInt(balanceThreshold),
+        }),
+      });
+      toast({
+        title: recurringSwaps ? "Saved successfully" : "Initiated swap",
```

Comment:
I got "initated swap" toast but the swap payment failed. We really need to fix the backend to do a proper sync payment

Reply:
Ah, from @im-adithya
> No, for swaps the HTLC is settled once we reveal the preimage to them, which happens when we broadcast a tx to claim from the lockup (which we do only after the lockup tx has been confirmed, hence taking 1 block time (or even more if fee increases). That's why we say swap "initiated" and not successful. How do you think we can update the user about the swap status?